extensions/amp-fit-text/0.1/test/test-amp-fit-text.js

```diff
@@ -53,6 +60,19 @@ describes.realWin(
         .then(() => ft);
     }
 
+    it('buildDom should have same DOM mutations as buildCallback', () => {
+      const fitText1 = createElementWithAttributes(doc, 'amp-fit-text', {
+        width: '111px',
+        height: '222px',
+      });
+      const fitText2 = fitText1.cloneNode(/* deep */ true);
+
+      new AmpFitText(fitText1).buildCallback();
+      buildDom(fitText2);
+
+      expect(fitText1.outerHTML).to.equal(fitText2.outerHTML);
+    });
```

Comment:
I wish there were a way to share this test case/make it easily reusable since it'll be needed everywhere, but I'm not sure what that would look like

Reply:
I was considering putting it all in a single file, within `src/compiler/test/test-buildDom`

Reply:
I like that idea, but it seemed to get more complicated when I thought through it. The helper would need some way of getting the `buildDom` function itself from the `tagName`, or it would have to be passed in, right? I was thinking the helper could take `element`, `Ctor`, and `buildDom` possibly

Reply:
You're right, I forgot about `buildDom`, it would need that too. So the test file would look like:
```js
const tests = [
  {tagName: 'amp-fit-text', buildDom: fitTextBuildDom, Ctor: AmpFitText },
  ...
]

tests.forEach(({Ctor, tagName, buildDom})=> {
  it('buildDom and buildCallback should result in the same outerHTML', () => {
     const el1 = createElementWithAttributes(tagName, {height: 10, width: 10});
     const el2 = el1.cloneNode(true);

     await (new Ctor(el1).buildCallback())
     buildDom(el2);
     expect(el1.outerHTML).to.equal(el2.outerHTML)
  })
})
```

Reply:
I disliked the idea when it was a helper function that different test files would import, but seeing it like this where all the components and functions/ctors are listed in one place, I like it much more. We can even do ``it(`${tagName} - buildDom..` => {...})`` so the test output is a bit more meaningful.

Reply:
hah, i asked about this about a moon ago. where should the file live?

Reply:
Is there a file anywhere that will be creating/exporting the instruction map? If so, it would make a lot of sense for this to be something like `test-instruction-map` and then the file itself would have a `const constructorMap` and compare the results

Reply:
Yes. Within `src/compiler/builders.js`

Reply:
Neither am I. Maybe we export a helper function that takes `tagName, Ctor` and itself uses the instruction map to look up the `buildDom` function, that way calling code can do something like `itMatchesBuildDom('amp-fit-text', AmpFitText);` or something like that